examples/tcp_option/README.md

```diff
@@ -0,0 +1,49 @@
+# TCP Header Modification Example
+This example focuses on modifying TCP headers, demonstrating how users can extend and manipulate network packet headers as required.
+
+## Building
+
+```shell
+$ cargo build
+```
+
+## Demo
```

Comment:
Having these instructions is great. Could we extend them further to be fully functional, though?

E.g., mention something like
```
nc --listen 127.0.0.1 65000 &
echo test | nc 127.0.0.1 65000
```
to trigger stuff?